sharding/nodesCoordinator/hashValidatorShuffler.go

```diff
@@ -288,16 +297,24 @@ func shuffleNodes(arg shuffleNodesArg) (*ResUpdateNodes, error) {
 		log.Warn("distributeValidators newNodes failed", "error", err)
 	}
 
-	err = arg.distributor.DistributeValidators(newWaiting, shuffledOutMap, arg.randomness, arg.flagBalanceWaitingLists)
-	if err != nil {
-		log.Warn("distributeValidators shuffledOut failed", "error", err)
+	if arg.flagStakingV4 {
+		err = distributeValidators(newWaiting, arg.auction, arg.randomness, false)
```

Comment:
I do not think that creating an extra function here would bring any benefit, since the only thing it would do is to call `distributeValidators` -> which already does its job.

I think the way to go here is to add an explanatory comment